### PR TITLE
feat(access-control): [6/6] add Cerbos provider + docs polish

### PR DESCRIPTION
**Pull request by amitgilad3**
## Summary
Part 6 of 6 (final), split out of #234. Stacked on #253.

Fully additive second policy backend behind the same `PolicyDecisionProvider` trait from part 2 (#250) — nothing else in the stack depends on this.

- New `CerbosProvider` implementing `PolicyDecisionProvider` against Cerbos's `CheckResources` API. Row filters/column masks have no native Cerbos concept, so they're carried through Cerbos's generic `outputs` mechanism using a documented `{"kind": "row_filter"|"column_mask", ...}` convention.
- Cerbos hard-requires non-empty `principal.roles` (an empty array is an HTTP 400 validation error, not a policy decision) — an identity with no roles is denied locally instead of surfacing an opaque provider error.
- `ProviderKind::Cerbos`, `CerbosProviderConfig`, and provider-aware dispatch in `AccessConnectionConfig::validate()`/`build_controller()`.
- `examples/with-cerbos/`: full Lakekeeper+Trino+Cerbos demo mirroring `examples/with-opa/`'s scenario.
- Docs: `cerbos.md`, provider-agnostic `overview.md`, provider logos, and new use-case pages (`internal-analysts.md`, `support-on-behalf-of.md`).

## Testing
40+ new tests: wire-format edge cases (multi-resource independence, array outputs, malformed/mixed outputs), an HTTP-stub provider suite (bearer auth, timeout, status/body mapping), and end-to-end guard tests (join with mixed allow/deny, combined filter+mask). `cargo check --workspace` passes on this branch (verified before pushing).

Once this merges, #234 can be closed — the union of all 6 PRs in this stack reproduces its diff against `main` exactly.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/queryflux-access-control/src/providers/cerbos/wire.rs [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+    pub roles: &'a [String],
+    pub attr: PrincipalAttr<'a>,
+}
+
+#[derive(Serialize)]
+pub(super) struct PrincipalAttr<'a> {
+    pub groups: &'a [String],
+    #[serde(flatten)]
+    pub attributes: &'a BTreeMap<String, Value>,
+}
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply crates/queryflux-access-control/src/providers/cerbos/wire.rs]: Fixed in 33a0e12d, with your suggested hand-written Serialize impl. `PrincipalAttr` now emits `groups` from the dedicated field and drops a same-named entry from the flattened `attributes` map, rather than emitting it twice. Added a regression test that inserts an `attributes["groups"]` entry and checks both the decoded JSON (one value, the field's) and the raw byte stream (exactly one `"groups":` occurrence, which is what Cerbos's decoder actually sees).

File: crates/queryflux-access-control/src/providers/cerbos/wire.rs [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+            let allow = result.actions.get(action).map(String::as_str) == Some(EFFECT_ALLOW);
+
+            let mut row_filters = Vec::new();
+            let mut column_masks = Vec::new();
+            for output in result.outputs {
+                let Ok(parsed) = serde_json::from_value::<PolicyOutputs>(output.val) else {
+                    // Not a row-filter/column-mask output (e.g. an unrelated audit
+                    // message) — not this provider's concern.
+                    continue;
+                };
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply crates/queryflux-access-control/src/providers/cerbos/wire.rs]: Fixed in 33a0e12d. A `row_filter`/`column_mask` output is now classified by its `kind` *before* attempting to parse the rest of the shape: an output with no recognized kind is still skipped (not ours), but one that names `row_filter`/`column_mask` and fails to parse now denies the resource with a reason, instead of silently vanishing. This also fixes the list case you flagged — I switched from deserializing the whole `val` as `PolicyOutputs` (whose untagged `Many` variant fails entirely if any list item doesn't match) to parsing each item individually. Added 4 tests: malformed column_mask, malformed row_filter, an unrecognized-kind output still skipped, and one malformed item denying even alongside a valid sibling in the same list.

File: crates/queryflux-core/src/access_config.rs [hunk 3]
```diff
@@ -138,61 +176,106 @@ impl Default for AccessConnectionConfig {
...
+                    format!("accessControl.connections.{name}.cerbos.url is not a valid URL: {e}")
+                })?;
+                match parsed.scheme() {
+                    "http" | "https" => {}
+                    other => {
+                        return Err(format!(
+                            "accessControl.connections.{name}.cerbos.url must be http or https, got {other}"
+                        ))
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply crates/queryflux-core/src/access_config.rs]: Fixed in 33a0e12d, per your suggested diff — same https-unless-loopback rule as `opa.clientCredentials.tokenEndpoint`, gated on whether `bearerToken` is actually set (self-hosted Cerbos with no token is unaffected). Added `cerbos_url_must_be_https_when_bearer_token_is_set_unless_loopback`.

File: examples/with-cerbos/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+      - POSTGRES_PASSWORD=postgres
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U postgres -p 5432 -d postgres"]
+      interval: 2s
+      timeout: 10s
+      retries: 10
+      start_period: 10s
+
+  minio:
+    image: minio/minio
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply examples/with-cerbos/docker-compose.yml]: Fixed in 33a0e12d — switched both services to `quay.io/minio/minio`, matching the convention `examples/full-stack` and `docker/docker-compose.yml` already use.

File: examples/with-cerbos/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        condition: service_healthy
+    restart: "no"
+    entrypoint: /bin/sh
+    command:
+      - -c
+      - >
+        curl -sf -X POST http://lakekeeper:8181/management/v1/bootstrap
+        -H 'Content-Type: application/json'
+        --data '{"accept-terms-of-use": true}';
+        exit 0
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply examples/with-cerbos/docker-compose.yml]: Fixed in 33a0e12d for both. Each now captures the HTTP status via `curl -s -o /tmp/resp -w '%{http_code}'` and only treats 2xx or 409 (already bootstrapped / already exists) as success; anything else prints the response body and exits 1, so `sentinel` can no longer become healthy before setup actually succeeded.

File: examples/with-cerbos/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    volumes:
+      - ./init-catalog.sql:/seed/init-catalog.sql:ro
+    depends_on:
+      trino:
+        condition: service_healthy
+      sentinel:
+        condition: service_healthy
+
+volumes:
+  queryflux-pg:
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply examples/with-cerbos/docker-compose.yml]: Fixed in 33a0e12d — added named volumes for `lakekeeper-db` and `minio`. I left Trino itself unmounted rather than guessing at its dynamic-catalog storage path (no other example in this repo persists that either, and getting a Trino internal directory wrong risks breaking startup); instead corrected the README to say plainly that Trino's dynamically-registered catalog is never persisted and `data-seed` must be re-run after any restart, not just a volume-wiped one.

File: examples/with-cerbos/init-catalog.sql [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    "s3.path-style-access" = 'true',
+    "s3.endpoint" = 'http://minio:9000',
+    "fs.native-s3.enabled" = 'true',
+    "s3.aws-access-key" = 'minio-root-user',
+    "s3.aws-secret-key" = 'minio-root-password'
+);
+
+CREATE SCHEMA IF NOT EXISTS lakekeeper.demo;
+
+DROP TABLE IF EXISTS lakekeeper.demo.customers;
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply examples/with-cerbos/init-catalog.sql]: Fixed in 33a0e12d, per your suggested direction. Rewrote both tables as `CREATE TABLE IF NOT EXISTS` + `INSERT ... WHERE NOT EXISTS (... WHERE id = fixture.id)`, so running it because one table is missing no longer drops or touches rows in the other, and re-running it when both already exist doesn't duplicate rows either. Left `seed.sql` untouched per your note — it's an explicit, separately-invoked reset.

File: website/docs/access-control/cerbos.md [hunk 8]
```diff
@@ -0,0 +1,415 @@
...
+3. Configure `sessionParamKeys: [customer_id]`.
+4. The policy checks that X may act for Y, then returns a row filter:
+
+```yaml
+- actions: ["table.select"]
+  roles: ["customer_portal"]
+  effect: EFFECT_ALLOW
+  condition:
+    match:
+      expr: R.attr.sessionParams.customer_id != ""
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply website/docs/access-control/cerbos.md]: Fixed in 33a0e12d. Added `sessionParams` to `ResourceAttr` (rendered as `R.attr.sessionParams.<key>`, matching the doc) and forward `req.context.session_params` into it from `to_request` — omitted entirely when empty rather than sent as `{}` on every request. Added two tests: one proving a session param reaches the wire payload, one proving it's omitted when none are configured. Went with implementing the feature rather than walking back the doc, since the plumbing already existed end-to-end (the guard already populates `RequestContext.session_params` from `sessionParamKeys` before calling the provider) — Cerbos's `to_request` was just the one place not reading it, the same gap already closed for OPA.

File: website/docs/access-control/internal-analysts.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+## Configuration sketch
+
+Humans authenticate; access control points at one named connection. Pick OPA or Cerbos for that connection — not both on the same name.
+
+```yaml
+auth:
+  provider: static          # or oidc — map groups/roles from the IdP
+  required: true
+  static:
+    users:
```
[the remark this answers is not in the document]
Comment (amitgilad3) [reply website/docs/access-control/internal-analysts.md]: Fixed in 33a0e12d for both files — `auth.staticUsers.users` as a username-keyed map with `password`, matching the real schema.
